ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryMetaDataCollector.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import org.joda.time.DateTime;
+
+public final class QueryMetaDataCollector {
+  private static final Hashtable<QueryIdKey, String> queryIdTable = new Hashtable<>();
+  private static final Hashtable<String, String> structurallySimilarTable = new Hashtable<>();
+
+  private QueryMetaDataCollector() {}
+
+  public static Map<String, String> getMetaData(
+      final KsqlConfig config,
+      final String nonAnonQuery,
+      final String anonQuery)
+      throws Exception {
+
+    // get metadata from control plane
+    final Map<String, String> metaData = new HashMap<>();
+    final String clusterNamespace = config
+        .getString(KsqlConfig.KSQL_CCLOUD_QUERYANONYMIZER_CLUSTER_NAMESPACE);
+
+    // check if metadata contains org id and cluster id
+    if (clusterNamespace.length() == 0) {
+      throw new Exception("Organization Id or Cluster Id could not be found");
+    }
```

Comment:
Don't throw, it will just means that all queries are treated as if they come from the same place (which makes sense for something like C3 deployment)

Reply:
We should think about getting C3 to understand these - you could default this config to be the KsqlConfig.KSQL_CLUSTER_NAME iirc